--- a/init	2025-12-24 12:17:03.768913145 +0800
+++ b/init	2025-12-24 12:17:51.917845067 +0800
@@ -314,7 +314,7 @@
 cd /
 # If there's no ROOT= specified, try to find one for 5 times
 if [ ! "$ROOT" ] || ! try_mount rw "$ROOT" /mnt; then
-	: "${ROOT_SEARCH_FILES:="fstab.android ${BOOT_IMAGE#/} system$SLOT.?fs system$SLOT.img system$SLOT/system/build.prop system/build.prop"}"
+	: "${ROOT_SEARCH_FILES:="system$SLOT.img fstab.android ${BOOT_IMAGE#/} system$SLOT.?fs system$SLOT/system/build.prop system/build.prop"}"
 
 	# If is live/install mode, prior to search for $INSTALL image, default to install.sfs
 	if [ "$INSTALL" ] || grep -q 'androidboot.live=true' /proc/cmdline; then
@@ -389,6 +389,11 @@
 setup_dpi
 post_detect
 post_setup_recovery
 
+if [ -n "$data_part" ];then
+	ln -s /dev/block/`basename $data_part` /dev/block/by-name/userdata
+	mount /dev/block/by-name/userdata /android/data
+fi
+
 if [ 0"$DEBUG" -gt 1 ]; then
 	echo -e "\nUse Alt-F1/F2/F3 to switch between virtual consoles"
